### DISCUSSION TITLE
Reading/parsing MET V10.0 ASCII2NC output in python

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#983] [File I/O] [answered]
I convert my little_r point observations text files to NetCDF for ingest into MET using MET ASCII2NC. Do you know if there is a python package or tool which will ingest/parse those NetCDF files so they can be used as input to other python tools?
I tried xarray, but it has errors when I tried to get it to read them.

**Comment by jprestop** [comment, marked as answer]
Hi John.  I would suggest using the netCDF4 Python interface: https://unidata.github.io/netcdf4-python/
You can view some [Sample Analysis Scripts](https://dtcenter.org/community-code/model-evaluation-tools-met/sample-analysis-scripts) that ingest/parse NetCDF files.  The examples use Python to pass data (ASCII, NetCDF, and Binary) to MET's plot_data_plane utility, which could be adapted to use as input to other Python tools.  Please take a look and let us know if you have any questions.

**Reply by jprestop** [reply]
Hi John.  I am attaching a screenshot of my screen.  Take a look at the very bottom where you'll see "Write a reply".  I hope this helps!
<img width="1363" alt="Screen Shot 2021-07-08 at 10 18 51 AM" src="https://user-images.githubusercontent.com/11999822/124957237-19932080-dfd6-11eb-8558-2cb972b6e3c8.png">

**Comment by jwraby** [comment]
CLASSIFICATION: UNCLASSIFIED

 

Hi Julie – 

 

Thanks for expanding on your suggestion. This is helpful.  I will take a look at the two sites to learn more about this approach.

 

R/

John

 

From: jprestop ***@***.***> 
Sent: Wednesday, July 7, 2021 9:26 AM
To: dtcenter/METplus ***@***.***>
Cc: Raby, John W CIV USARMY DEVCOM ARL (USA) ***@***.***>; Author ***@***.***>
Subject: [Non-DoD Source] Re: [dtcenter/METplus] Reading/parsing MET V10.0 ASCII2NC output in python (#983)

 

All active links contained in this email were disabled. Please verify the identity of the sender, and confirm the authenticity of all links contained within the message prior to copying and pasting the address to a Web browser. 

  _____  

 

Hi John. I would suggest using the netCDF4 Python interface: Caution-https://unidata.github.io/netcdf4-python/ < Caution-https://unidata.github.io/netcdf4-python/ > 
You can view some Sample Analysis Scripts < Caution-https://dtcenter.org/community-code/model-evaluation-tools-met/sample-analysis-scripts >  that ingest/parse NetCDF files. The examples use Python to pass data (ASCII, NetCDF, and Binary) to MET's plot_data_plane utility, which could be adapted to use as input to other Python tools. Please take a look and let us know if you have any questions.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub < Caution-https://github.com/dtcenter/METplus/discussions/983#discussioncomment-975430 > , or unsubscribe < Caution-https://github.com/notifications/unsubscribe-auth/AUX5FH52HGHGYVNGZJLADOLTWRWWZANCNFSM475M3VIQ > .  <Caution-https://github.com/notifications/beacon/AUX5FHYGJ3UQ5X2D6B64J3LTWRWWZA5CNFSM475M3VI2YY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAAHOERQ.gif> 

CLASSIFICATION: UNCLASSIFIED

**Comment by jwraby** [comment]
CLASSIFICATION: UNCLASSIFIED

 

Julie – 

 

I didn’t see a way to append more comments to this thread in GitHub, so I’m sending this reply to let you know that I think I successfully opened a NetCDF point observation file (MET V9.1 ASCII2NC output) using python. I didn’t see the errors I was getting before and I saw the observations metadata displayed on the screen instead. Thanks for the tip on using the netcdf4 interface. I installed netcdf4 and some other related (dependencies?) packages which enabled xarray to open the file. 

 

How can I add follow-up comments to the thread on GitHub?

 

R/

John

 

From: Raby, John W CIV USARMY DEVCOM ARL (USA) 
Sent: Wednesday, July 7, 2021 9:30 AM
To: 'dtcenter/METplus' ***@***.***>
Subject: RE: [Non-DoD Source] Re: [dtcenter/METplus] Reading/parsing MET V10.0 ASCII2NC output in python (#983) (UNCLASSIFIED)

 

CLASSIFICATION: UNCLASSIFIED

 

Hi Julie – 

 

Thanks for expanding on your suggestion. This is helpful.  I will take a look at the two sites to learn more about this approach.

 

R/

John

 

From: jprestop ***@***.*** ***@***.***> > 
Sent: Wednesday, July 7, 2021 9:26 AM
To: dtcenter/METplus ***@***.*** ***@***.***> >
Cc: Raby, John W CIV USARMY DEVCOM ARL (USA) ***@***.*** ***@***.***> >; Author ***@***.*** ***@***.***> >
Subject: [Non-DoD Source] Re: [dtcenter/METplus] Reading/parsing MET V10.0 ASCII2NC output in python (#983)

 

All active links contained in this email were disabled. Please verify the identity of the sender, and confirm the authenticity of all links contained within the message prior to copying and pasting the address to a Web browser. 

  _____  

 

Hi John. I would suggest using the netCDF4 Python interface: Caution-https://unidata.github.io/netcdf4-python/ < Caution-https://unidata.github.io/netcdf4-python/ > 
You can view some Sample Analysis Scripts < Caution-https://dtcenter.org/community-code/model-evaluation-tools-met/sample-analysis-scripts >  that ingest/parse NetCDF files. The examples use Python to pass data (ASCII, NetCDF, and Binary) to MET's plot_data_plane utility, which could be adapted to use as input to other Python tools. Please take a look and let us know if you have any questions.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub < Caution-https://github.com/dtcenter/METplus/discussions/983#discussioncomment-975430 > , or unsubscribe < Caution-https://github.com/notifications/unsubscribe-auth/AUX5FH52HGHGYVNGZJLADOLTWRWWZANCNFSM475M3VIQ > .  <Caution-https://github.com/notifications/beacon/AUX5FHYGJ3UQ5X2D6B64J3LTWRWWZA5CNFSM475M3VI2YY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAAHOERQ.gif> 

CLASSIFICATION: UNCLASSIFIED

CLASSIFICATION: UNCLASSIFIED

**Reply by jprestop** [reply]
Hi John.  I'm so glad you had success with the netCDF4 Python interface!  Thanks for letting us know.  I think the best way to add follow up comments is to reply to the Answer (as I am doing here).  At the bottom of this Answer and the others above you should see a little text box at the bottom that says "Write a reply", where you can type.   Is that what you're looking for?  I hope that helps.

**Reply by jwraby** [reply]
Hi Julie - I'm now viewing the thread (#983) after logging in to my account on GitHub. I found the text box where I composed this reply to you. Before when I had trouble seeing the text box, I was viewing the same thread on GitHub seeing the original question and the now several replies, while no logged in. It was a bit confusing to be able to see everything just like I was logged in while not actually being logged in. When I did log in, I immediately saw the text box where you said it would be. I have to remember to always log in to GitHub!

**Reply by jprestop** [reply]
Great!  Thanks for letting us know that you found it and that the issue was that you were not logged in.  That's great feedback for us to have for other users as we navigate this new way of providing support.

**Reply by jwraby** [reply]
Glad to help out. I originally navigated to the GitHub site this morning using my Chrome browser by selecting a bookmark for the site. It went there where, for all intents and purposes, it appeared just like I was logged in. I saw all the posted info, question and replies, but not the functionality to "Write a reply". All my subsequent logins during the day were all automatically logged in as I had never logged out.

**Comment by jwraby** [comment]
CLASSIFICATION: UNCLASSIFIED

 

Julie – 

 

I’m logged in to GitHub using the URL in your email below:

https://github.com/dtcenter/METplus/discussions/983#discussioncomment-980052

 

I’m having trouble seeing the text box you mention for “Write a reply".

 

Can you be more specific on the location of the text box?

 

Thanks.

 

R/

John

 

From: jprestop ***@***.***> 
Sent: Thursday, July 8, 2021 9:05 AM
To: dtcenter/METplus ***@***.***>
Cc: Raby, John W CIV USARMY DEVCOM ARL (USA) ***@***.***>; Author ***@***.***>
Subject: [Non-DoD Source] Re: [dtcenter/METplus] Reading/parsing MET V10.0 ASCII2NC output in python (#983)

 

All active links contained in this email were disabled. Please verify the identity of the sender, and confirm the authenticity of all links contained within the message prior to copying and pasting the address to a Web browser. 

  _____  

 

Hi John. I'm so glad you had success with the netCDF4 Python interface! Thanks for letting us know. I think the best way to add follow up comments is to reply to the Answer (as I am doing here). At the bottom of this Answer and the others above you should see a little text box at the bottom that says "Write a reply", where you can type. Is that what you're looking for? I hope that helps.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub < Caution-https://github.com/dtcenter/METplus/discussions/983#discussioncomment-980052 > , or unsubscribe < Caution-https://github.com/notifications/unsubscribe-auth/AUX5FH33BSLJ3JWT4YL6BDLTWW5BBANCNFSM475M3VIQ > .  <Caution-https://github.com/notifications/beacon/AUX5FH2HUJKEIGVYDJC2XP3TWW5BBA5CNFSM475M3VI2YY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAAHPIVA.gif> 

CLASSIFICATION: UNCLASSIFIED
